src/SAML2/AuthnRequest.php

```diff
@@ -307,15 +305,40 @@ public function buildRequestQuery()
      */
     public function getSignedRequestQuery()
```

Comment:
As discussed, we can mitigate this risk by checking whether the HTTP query parameters contain only one `SAMLRequest`, `Signature` and `SigAlg` and zero or one `RelayState`. We should throw exceptions if this is not the case. To ensure this behaviour (and document it through code), tests should be added. Other parameters should be left alone as per the [SAML Bindings specification](https://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf), see r. 566-567, page 16.